pkg/cloud/services/compute/instance_types.go

```diff
@@ -116,7 +116,6 @@ func (is *InstanceStatus) BastionStatus(openStackCluster *infrav1.OpenStackClust
 
 	clusterNetwork := openStackCluster.Status.Network.Name
 	i.IP = ns.IP(clusterNetwork)
-	i.FloatingIP = ns.FloatingIP(clusterNetwork)
```

Comment:
Here we were persisting the floating IP in status only if it was attached to the bastion VM. In order to properly track the created floating IP, it is now [saved independently in reconcileBastion](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/398e4439f4a82649e1c6e4232d08c275e4b2c189/controllers/openstackcluster_controller.go#L421)